scripts/run_lighthouse_a11y_tests.py

```diff
@@ -0,0 +1,174 @@
+    puppeteer_path = os.path.join(
+        'core', 'tests', 'puppeteer', 'lighthouse_setup.js')
+    bash_command = [node_path, puppeteer_path]
+
+    try:
+        script_output = subprocess.check_output(bash_command).split('\n')
+        python_utils.PRINT(script_output)
+        for url in script_output:
+            export_url(url)
+        python_utils.PRINT(
+            'Puppeteer script completed successfully.')
+
+    except subprocess.CalledProcessError:
+        python_utils.PRINT(
+            'Puppeteer script failed. More details can be found above.')
+        sys.exit(1)
+
+
+def export_url(url):
+    """Exports the url to an environmental variable."""
+    url_list = url.split('/')
+    if 'collection_editor' in url:
+        os.environ['collection_editor'] = url_list[5]
+    elif 'create' in url:
+        os.environ['exploration_editor'] = url_list[4]
+    elif 'topic_editor' in url:
+        os.environ['topic_editor'] = url_list[4]
+    elif 'story_editor' in url:
+        os.environ['story_editor'] = url_list[4]
+    elif 'skill_editor' in url:
+        os.environ['skill_editor'] = url_list[4]
+    else:
+        return
+
+
+def run_lighthouse_checks():
+    """Runs the lighthouse checks through the .lighthouserc.js config."""
+
+    node_path = os.path.join(common.NODE_PATH, 'bin', 'node')
+    lhci_path = os.path.join('node_modules', '@lhci', 'cli', 'src', 'cli.js')
+    bash_command = [
+        node_path, lhci_path, 'autorun', '--config=.lighthouserc-a11y.js']
+
+    try:
+        subprocess.check_call(bash_command)
+        python_utils.PRINT('Lighthouse checks completed successfully.')
+    except subprocess.CalledProcessError:
+        python_utils.PRINT(
+            'Lighthouse checks failed. More details can be found above.')
+        sys.exit(1)
+
+
+def enable_webpages():
+    """Enables deactivated webpages for testing."""
+
+    pattern = 'CONTRIBUTOR_DASHBOARD_ENABLED = .*'
+    replace = 'CONTRIBUTOR_DASHBOARD_ENABLED = True'
+    common.inplace_replace_file(FECONF_FILE_PATH, pattern, replace)
+
+    pattern = '"ENABLE_ACCOUNT_DELETION": .*'
+    replace = '"ENABLE_ACCOUNT_DELETION": true,'
+    common.inplace_replace_file(CONSTANTS_FILE_PATH, pattern, replace)
+
+
+def start_google_app_engine_server():
+    """Start the Google App Engine server."""
+
+    app_yaml_filepath = 'app_dev.yaml'
+    p = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' %
+        (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_SDK_HOME,
+            GOOGLE_APP_ENGINE_PORT, app_yaml_filepath
+        ), shell=True)
+    SUBPROCESSES.append(p)
+
+
+def main():
+    """Runs lighthouse checks and deletes reports."""
+
+    enable_webpages()
+    atexit.register(cleanup)
+
+    build.main(args=[])
+    background_processes = []
```

Comment:
How is this list used in this script?

Reply:
I'm not entirely sure. I know that its appended by the subprocess below. But I was referencing start.py at line 141 on how to start the oppia server. Could you check if I'm starting it properly?

Reply:
Edit* After using the run_webpack_compilation() function from run_e2e_tests.py file, I don't seem to need background_processes